### PR TITLE
Permissions - Roles

### LINKED ISSUE #24: Permissions - Roles
Allow user to have a role, default is parent
Roles:
- Parents
- Children
- Guests
- Strangers

### PR DESCRIPTION
**Pull request by Karin-kazarian**
Resolve #24

### CODE REVIEW

File: SmartHomeSimulatorBack/SmartHomeSimulator/src/main/java/com/project/SmartHomeSimulator/model/User.java [hunk 1]
```diff
@@ -17,6 +18,9 @@
     private String username;
     @NotBlank(message = "Password cannot be blank")
     private String password;
+    @NotNull(message = "Role cannot be blank")
+    @Enumerated(EnumType.STRING)
```
Comment (willyyhuang) [inline SmartHomeSimulatorBack/SmartHomeSimulator/src/main/java/com/project/SmartHomeSimulator/model/User.java:22]: love this

**Review by willyyhuang — APPROVED** [review]

File: SmartHomeSimulatorBack/SmartHomeSimulator/src/main/java/com/project/SmartHomeSimulator/model/Role.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.project.SmartHomeSimulator.model;
+
+public enum Role {
+    PARENT,CHILD,GUEST,STRANGER
```
[the remark this answers is not in the document]
Comment (jenniferasmar) [reply SmartHomeSimulatorBack/SmartHomeSimulator/src/main/java/com/project/SmartHomeSimulator/model/Role.java:4]: Parent = admin

**Review by jenniferasmar — APPROVED** [review]

**Review by SekayZ — APPROVED** [review]
Looks good

**Review by hodansh — COMMENTED** [review]
Cool!

**Pull request merged**